entrypoint.sh

```diff
@@ -85,27 +118,29 @@ then
   execute_commands "$INPUT_PRECOMMANDS"
 fi
 
-# If an environment is detected as input, for each secret specified get the value of
-# the matching named environment variable then configure using wrangler secret put.
-# Skip if publish is set to false.
-if [ "$INPUT_PUBLISH" != "false" ]
-then
-  if [ -z "$INPUT_ENVIRONMENT" ]
-  then
-    wrangler publish
+# If we have secrets, set them
+for SECRET in $INPUT_SECRETS; do
+  VALUE=$(printenv "$SECRET") || secret_not_found "$SECRET"
 
-    for SECRET in $INPUT_SECRETS; do
-      VALUE=$(printenv "$SECRET") || secret_not_found "$SECRET"
-      echo "$VALUE" | wrangler secret put "$SECRET"
-    done
+  if [ -z "$INPUT_ENVIRONMENT" ]; then
+    echo "$VALUE" | wrangler secret put "$SECRET"
   else
-    wrangler publish -e "$INPUT_ENVIRONMENT"
+    echo "$VALUE" | wrangler secret put "$SECRET" --env "$INPUT_ENVIRONMENT"
+  fi
+done
 
-    for SECRET in $INPUT_SECRETS; do
-      VALUE=$(printenv "$SECRET") || secret_not_found "$SECRET"
-      echo "$VALUE" | wrangler secret put "$SECRET" --env "$INPUT_ENVIRONMENT"
-    done
+# If there's no input command then default to publish otherwise run it
+if [ -z "$INPUT_COMMAND" ]; then
+  echo "::notice:: No command was provided, defaulting to 'publish'"
+
+ if [ -z "$INPUT_ENVIRONMENT" ]; then
+    wrangler publish
+  else
+    wrangler publish --env "$INPUT_ENVIRONMENT"
   fi
+
+else
+  execute_commands "wrangler $INPUT_COMMAND"
```

Comment:
Open question: I don't think it's feasible for me to pass --env here as the user may also do that or it may result in different behaviour than intended (or even an unknown option warning/error).

How do we handle the env here?

Reply:
Perhaps throw an error if both INPUT_COMMAND and INPUT_ENVIRONMENT are set, saying that if you are setting your own command then you must pass the environment as a command line arg?

Reply:
I can't tell if the command has --env but we still need INPUT_ENVIRONMENT for secrets. Erroring on both existing wouldn't work since that's ok, I can add a notice but it may be a little annoying for those who do have it setup correctly. Will add a notice for now